concordium-consensus/src/Concordium/KonsensusV1/Consensus/Blocks.hs

```diff
@@ -1371,3 +1377,28 @@ makeBlock = do
             void . onTimeout (DelayUntil . timestampToUTCTime $ blockTimestamp block) $ do
                 sendBlock block
                 checkedValidateBlock block
+
+-- | Compute the missed rounds for each validator. Starts from the given parent
+--   block up to (not including) the given final round.
+computeMissedRounds ::
+    forall m.
+    (BlockState m ~ HashedPersistentBlockState (MPV m), BlockStateQuery m) =>
+    Option TimeoutCertificate ->
+    BlockPointer (MPV m) ->
+    Round ->
+    m [(BakerId, Word16)]
+computeMissedRounds tc _parent _rnd | isAbsent tc = return []
+computeMissedRounds _tc parent rnd = do
+    let parentBlockState = bpState parent
+    validators <- getCurrentEpochBakers parentBlockState
+    seedState <- getSeedState parentBlockState
```

Comment:
Also, I think the functionality should be moved to the `LeaderElection` module.

Reply:
Moved it to `LeaderElection` and made it pure. `seedState` still needs to be computed at the call site however.

Reply:
Shouldn't the parent block and the child block be in the same epoch regardless. Because we know that there exists a timeout certificate for the parent block, and hence an epoch transition wasn't possible from parent to child block.

Reply:
Suppose we have block 1 in round r, epoch e which is a trigger block for the epoch transition. And we have block 2 in (r+1,e), with parent 1. When we have a QC for block 2, then block 1 is finalized and we can transition to epoch e+1. If round r+2 then times out, block 3 can be in (r+3,e+1) with parent 2.

Reply:
You're right. I fixed this, `computeMissedRounds` now takes the leader ship election nonce (of the epoch of the child block) as argument. This way the seed state doesn't have to be computed anymore either.